sklearn/tree/_tree.pyx

```diff
@@ -138,6 +138,10 @@ cdef class Criterion:
 
         pass
 
+    cdef void reverse_reset(self) nogil:
+        """Reset the criterion at pos=end."""
+        pass
```

Comment:
Actually I am not sure whether we should add such a method or pass a flag instead in `reset`? (I have not strong opinion about one or the other. Though, flat might be better than nested)